### PR TITLE
Clarify template configuration documentation

### PR DESCRIPTION
**Pull request by onlyhavecans**
I couldn't figure out how to use templates and where they were written. This makes how they are written, when they are written, and how to use them much clearer in the reference documentation. This was a major annoying hurdle for me until @robbkidd showed me how it works and then helped document it.

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: www/source/partials/docs/_dev-pkgs-add-configuration.html.md.erb [hunk 1]
```diff
@@ -16,21 +16,21 @@ We can make this user tunable like this:
 recv_buffer {{cfg.recv_buffer}}
 ```
 
-Habitat can read values that it will use to render the templatize
-config files in three ways:
+Habitat can read values that it will use to render the templatize config files in three ways:
```
Comment (robbkidd) [inline www/source/partials/docs/_dev-pkgs-add-configuration.html.md.erb]: `s/templatize/templatized/`?

File: www/source/partials/docs/_dev-pkgs-add-configuration.html.md.erb [hunk 2]
```diff
@@ -16,21 +16,21 @@ We can make this user tunable like this:
...
 
 1. `default.toml` - Each plan includes a `default.toml` file that specifies the default values to use in the absence of any user provided inputs. These files are written in [TOML](https://github.com/toml-lang/toml), a simple config format.
-2. Environment variable - At start up, tunable config values can be passed to Habitat using environment variables.
-3. At runtime - Users can alter config at runtime using `hab config
-apply`. The input for this command also uses the TOML format.
+2. At runtime - Users can alter config at runtime using `hab config apply`. The input for this command also uses the TOML format.
+3. Environment variable - At start up, tunable config values can be passed to Habitat using environment variables; this most over-riding way of setting these but require you to restart the supervisor to change them.
```
Comment (robbkidd) [inline www/source/partials/docs/_dev-pkgs-add-configuration.html.md.erb]: Yes, I believe these priorities and caveats about use are still true.

**Comment by tashimi** [comment]
@thesentinels approve

tenor-109595935

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/338742247).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/338742247).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**